app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -653,11 +653,11 @@ private void showText(String title, String body) {
             String message = "";
 
             if (title != null) {
-                if (title.length() > 12) {
+                if (title.length() > 17) {
                     message = title.substring(0, 12);
```

Comment:
AFAIU you're trying to create a message that is exactly n characters long (so that it fits onto the display. Maybe we should have a method (or two) in a class StringUtils (or similar) that does this:
truncate(String, numCharacters)
pad(String, numCharacters)